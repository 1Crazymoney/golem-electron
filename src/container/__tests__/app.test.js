jest.unmock('../App')

import React from 'react'
import ReactDOM from 'react-dom'

import { Provider } from 'react-redux'
import { Router, Route, createMemoryHistory } from "react-router";

import TestUtils from 'react-addons-test-utils'
import configureStore from 'redux-mock-store';
import ConnectedApp, { App } from '../App'
import sinon from 'sinon'

const mockStore = configureStore({})
const history = createMemoryHistory("/test")
const wrapper = shallow(
    <Provider store={ mockStore({})}>
            <ConnectedApp history={history}/>
        </Provider>
)
it('should render an App container', () => {
    const wrapper = shallow(
        <Provider store={ mockStore({})}>
            <ConnectedApp/>
        </Provider>
    )
    expect(wrapper).toMatchSnapshot()
})

it('should check history prop', () => {
    const wrapper = shallow(
        <Provider store={ mockStore({})}>
            <ConnectedApp history={history}/>
        </Provider>
    )
    expect(wrapper.props().history).toEqual(history)
})


describe('<App />', () => {
    const actions = {
        login: () => true
    }
    const wrapper = shallow(
        <App history={history} actions={actions}/>
    )
    it('renders one <Router /> component', () => {
        expect(wrapper.find(Router).length).toBe(1)
    });

    it('renders four <Route /> components', () => {
        expect(wrapper.find(Route).length).toBe(4)
    });

    it('should call componentDidMount', () => {
        sinon.spy(App.prototype, 'componentDidMount');
        expect(App.prototype.componentDidMount.calledOnce).toBe(false)
        const wrapper = mount(<App history={history} actions={actions}/>);
        expect(App.prototype.componentDidMount.calledOnce).toBe(true)
    })
});