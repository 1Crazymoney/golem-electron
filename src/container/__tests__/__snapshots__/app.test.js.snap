exports[`test should render an App container 1`] = `<Connect(App) />`;
